tests/Models/Comment.php

```diff
@@ -15,15 +15,15 @@ class Comment extends Model
     use HasTableAlias;
 
     /**
-     * @return \Znck\Eloquent\Relations\BelongsToThrough<\Tests\Models\Country, \Tests\Models\User|\Tests\Models\Post, $this>
+     * @return \Znck\Eloquent\Relations\BelongsToThrough<\Tests\Models\Country, list<\Tests\Models\User|\Tests\Models\Post>, $this>
```

Comment:
Try this instead:

```suggestion
     * @return \Znck\Eloquent\Relations\BelongsToThrough<\Tests\Models\Country, array{0: \Tests\Models\User|\Tests\Models\Post}, $this>
```

Reply:
I believe your suggestion would be better, but the current generics don't support this yet:

```
 Line   tests/Models/Comment.php                                                                 
 ------ ----------------------------------------------------------------------------------------- 
  22     Method Tests\Models\Comment::country() should return                                     
         Znck\Eloquent\Relations\BelongsToThrough<Tests\Models\Country, array{Tests\Models\User,  
         Tests\Models\Post}, $this(Tests\Models\Comment)> but returns                             
         Znck\Eloquent\Relations\BelongsToThrough<Tests\Models\Country, array<int,                
         Tests\Models\Post|Tests\Models\User>, $this(Tests\Models\Comment)>.                      
         🪪  return.type   
```